docs/the-new-architecture/using-codegen.md

```diff
@@ -0,0 +1,199 @@
+# Using Codegen
+
+This guide teaches how to configure **Codegen**, and how to invoke it manually for each platform, and describes the generated code.
+
+## Prerequisites
+
+You always need a React Native app to generate the code properly, even when invoking the **Codegen** manually.
+
+The **Codegen** process is tightly coupled with the build of the app, and the scripts are located in the `react-native` NPM package.
+
+For the sake of this guide, create a project using the React Native CLI as follows:
+
+```bash
+npx @react-native-community/cli@latest init SampleApp --version 0.76.0
+```
+
+**Codegen** is used to generate the glue-code for your custom modules or components. See the guides for Turbo Native Modules and Fabric Native Components for more details on how to create them.
+
+<!-- TODO: add links -->
+
+## Configuring **Codegen**
+
+**Codegen** can be configured in your app by modifying the `package.json` file. **Codegen** is controlled by a custom field called `codegenConfig`.
+
+```json title="package.json"
+  "codegenConfig": {
+    "name": "<SpecName>",
+    "type": "<types>",
+    "jsSrcsDir": "<source_dir>",
+    "android": {
+      "javaPackageName": "<java.package.name>"
+    }
+  },
+```
+
+You can add this snippet to your app and customise the various fields:
+
+- `name:` This is the name that will be used to create files containig your specs. The convention is that it should have the suffix `Spec`, but this is not mandatory.
+- `type`: the type of code we need to generate. Allowed values are `modules`, `components`, `all`.
+  - `modules:` use this value if you only need to generate code for Turbo Native Modules.
+  - `components:` use this value if you only need to generate code for Native Fabric Components.
+  - `all`: use this value if you have a mixture of components and modules.
+- `jsSrcsDir`: this is the root folder where all your specs live.
+- `android.javaPackageName`: this is an android specific setting to let **Codegen** generate the files in a custom package.
+
+When **Codegen** runs, it searches among all the dependencies of the app, looking for JS files that respects some specific conventions, and it generates the required code:
+
+- Turbo Native Modules requires that the spec files are prefixed with `Native`. For example, `NativeLocalStorage.ts` is a valid name for a spec file.
+- Native Fabric Components requires that the spec files are suffixed with `NativeComponent`. For example, `WebViewNativeComponent.tx` is a valid name for a spec file.
+
+## Running **Codegen**
+
+The rest of this guide assumes that you have a Native Turbo Module and/or a Native Fabric Component already set up in your project. We also assume that you have valid specification files in the `jsSrcsDir` specified in the `package.json`.
```

Comment:
```suggestion
The rest of this guide assumes that you have a Native Turbo Module, Native Fabric Component or both already set up in your project. We also assume that you have valid specification files in the `jsSrcsDir` specified in the `package.json`.
```
Strangely this comes up with Shrunk & White and the suggestion is to do the above.